In [124]:
import pandas as pd

In [125]:
 # Load CSV file into DataFrame
df = pd.read_csv("../Data/credit_risk_dataset.csv") 
df.head()  # Show first 5 rows

,person_age,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length
0,22,59000,RENT,123.00,PERSONAL,D,35000,16.02,1,0.59,Y,3
1,21,9600,OWN,5.00,EDUCATION,B,1000,11.14,0,0.10,N,2
2,25,9600,MORTGAGE,1.00,MEDICAL,C,5500,12.87,1,0.57,N,3
3,23,65500,RENT,4.00,MEDICAL,C,35000,15.23,1,0.53,N,2
4,24,54400,RENT,8.00,MEDICAL,C,35000,14.27,1,0.55,Y,4


In [126]:
# Print dataset shape (rows, columns)
print(f'Rows X Columns: {df.shape}')

Rows X Columns: (32581, 12)


In [127]:
# Show summary of dataset (columns, data types, missing values)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32581 entries, 0 to 32580
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   person_age                  32581 non-null  int64  
 1   person_income               32581 non-null  int64  
 2   person_home_ownership       32581 non-null  object 
 3   person_emp_length           31686 non-null  float64
 4   loan_intent                 32581 non-null  object 
 5   loan_grade                  32581 non-null  object 
 6   loan_amnt                   32581 non-null  int64  
 7   loan_int_rate               29465 non-null  float64
 8   loan_status                 32581 non-null  int64  
 9   loan_percent_income         32581 non-null  float64
 10  cb_person_default_on_file   32581 non-null  object 
 11  cb_person_cred_hist_length  32581 non-null  int64  
dtypes: float64(3), int64(5), object(4)
memory usage: 3.0+ MB


Some data types here are incorrect, need to change.

In [128]:
# Count missing values in each column
df.isna().sum()

person_age                       0
person_income                    0
person_home_ownership            0
person_emp_length              895
loan_intent                      0
loan_grade                       0
loan_amnt                        0
loan_int_rate                 3116
loan_status                      0
loan_percent_income              0
cb_person_default_on_file        0
cb_person_cred_hist_length       0
dtype: int64

Columns "person_emp_length" and "loan_int_rate" have missing values ​​that need to be handled.

In [129]:
print(f"Duplicate values: {df.duplicated().sum()}")

Duplicate values: 165


In [130]:
# Show statistical summary of numerical columns
df.describe()

,person_age,person_income,person_emp_length,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_cred_hist_length
count,32581.00,32581.00,31686.00,32581.00,29465.00,32581.00,32581.00,32581.00
mean,27.73,66074.85,4.79,9589.37,11.01,0.22,0.17,5.80
std,6.35,61983.12,4.14,6322.09,3.24,0.41,0.11,4.06
min,20.00,4000.00,0.00,500.00,5.42,0.00,0.00,2.00
25%,23.00,38500.00,2.00,5000.00,7.90,0.00,0.09,3.00
50%,26.00,55000.00,4.00,8000.00,10.99,0.00,0.15,4.00
75%,30.00,79200.00,7.00,12200.00,13.47,0.00,0.23,8.00
max,144.00,6000000.00,123.00,35000.00,23.22,1.00,0.83,30.00


Age
- Average age is 27.73 years
- Minimum is 20, but maximum is 144, which looks like an outlier

Income 
- Mean income is 66,074
- Maximum value (6,000,000) is very high → possible extreme outlier

Employment Length 
- Average work experience is 4.79 years
- Maximum value is 123, which is unrealistic → needs outlier handling

The other columns are fine. They need to be adjusted as needed.

In [131]:
# Count of each class in loan_status (0 = paid, 1 = default)
df['loan_status'].value_counts()

loan_status
0    25473
1     7108
Name: count, dtype: int64

In target column, classes are unbalanced. These need to be handled.

In [132]:
sel_col = ['person_home_ownership', 'loan_intent', 'loan_grade', 'cb_person_default_on_file']
for col in sel_col:
    print(df[col].value_counts())
    print("-----------------------")

person_home_ownership
RENT        16446
MORTGAGE    13444
OWN          2584
OTHER         107
Name: count, dtype: int64
-----------------------
loan_intent
EDUCATION            6453
MEDICAL              6071
VENTURE              5719
PERSONAL             5521
DEBTCONSOLIDATION    5212
HOMEIMPROVEMENT      3605
Name: count, dtype: int64
-----------------------
loan_grade
A    10777
B    10451
C     6458
D     3626
E      964
F      241
G       64
Name: count, dtype: int64
-----------------------
cb_person_default_on_file
N    26836
Y     5745
Name: count, dtype: int64
-----------------------


In [133]:
#duplicate values remove
df = df.drop_duplicates()

In [134]:
print('Age > 80:')
df[df['person_age'] > 80]

Age > 80:


,person_age,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length
81,144,250000,RENT,4.00,VENTURE,C,4800,13.57,0,0.02,N,3
183,144,200000,MORTGAGE,4.00,EDUCATION,B,6000,11.86,0,0.03,N,2
575,123,80004,RENT,2.00,EDUCATION,B,20400,10.25,0,0.25,N,3
747,123,78000,RENT,7.00,VENTURE,B,20000,NaN,0,0.26,N,4
32297,144,6000000,MORTGAGE,12.00,PERSONAL,C,5000,12.73,0,0.00,N,25
32416,94,24000,RENT,1.00,MEDICAL,C,6500,NaN,0,0.27,N,27
32506,84,94800,MORTGAGE,2.00,PERSONAL,A,10000,7.51,0,0.11,N,24


It's unnatural to look over 80. These need to be thrown away.

In [135]:
# Age over 80, romove these rows
df = df[df['person_age'] < 80]

In [138]:
print("Employee Who work >60 year")
df[df['person_emp_length'] > 60]

Employee Who work >60 year


,person_age,person_income,person_home_ownership,person_emp_length,loan_intent,loan_grade,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_default_on_file,cb_person_cred_hist_length
0,22,59000,RENT,123.00,PERSONAL,D,35000,16.02,1,0.59,Y,3
210,21,192000,MORTGAGE,123.00,VENTURE,A,20000,6.54,0,0.10,N,4


It is also unusual to see more than 60 years of work experience. These need to be removed.

In [139]:
df = df[df['person_emp_length'] < 60]